code/modules/hydroponics/hydroponics.dm

```diff
@@ -1243,6 +1268,17 @@
 	plant_health = add_output_port("Plant Health", PORT_TYPE_NUMBER)
 	reagents_level = add_output_port("Reagents Level", PORT_TYPE_NUMBER)
 
+/obj/machinery/hydroponics/Entered(atom/movable/arrived, atom/old_loc, list/atom/old_locs)
+	. = ..()
+	if(!istype(arrived, /obj/item/clothing/head/mob_holder/snail))
+		return
+	our_snail = new
+	vis_contents += our_snail
+	our_snail.layer = layer + 0.01
+	var/obj/item/clothing/head/mob_holder/snail = arrived
+	RegisterSignal(snail.held_mob, COMSIG_LIVING_DEATH, PROC_REF(on_snail_death)) //rip
```

Comment:
Looks like you don't handle exited or qdeleted here
At minimum becuase you keep a hard ref to the snail you will need to clear it if the snail holder is deleted, and probably also if it exits this loc

If someone cast a return spell on a snail and put it in a hydro tray they could put the same snail in every tray right now, I think.